compiler/next/lib/AST/Decl.cpp

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "chpl/AST/Decl.h"
+
+#include "chpl/AST/Expr.h"
+
+namespace chpl {
+namespace ast {
+
+Decl::Decl(asttags::ASTTag tag, owned<Symbol> symbol)
+  : Expr(tag, makeASTList(std::move(symbol))) {
+}
+
+Decl::~Decl() {
+}
```

Comment:
This should not be here because the destructor is declared `=0`.

Reply:
My understanding of the C++ is that we need to define even these pure virtual destructors. I do not have an explanation for why.

Reply:
E.g. from the first answer in https://stackoverflow.com/questions/1219607/why-do-we-need-a-pure-virtual-destructor-in-c

> Note: The destructor is the only method that even if it is pure virtual has to have an implementation in order to instantiate derived classes (yes pure virtual functions can have implementations).